File: largest_possible_remainder.rb
#!/usr/bin/env ruby
# http://programmingpraxis.com/2015/10/16/largest-possible-remainder/

def largest_possible_remainder(n,d=n)
  div_with_largest_remainder = 1
  (2..d).each { |div| div_with_largest_remainder = div if n % div > n % div_with_largest_remainder }
  [div_with_largest_remainder,n%div_with_largest_remainder]
end

p largest_possible_remainder(20,10)

File: largest_possible_remainder.swift
#!/usr/bin/env swift
// http://programmingpraxis.com/2015/10/16/largest-possible-remainder/

func largestPossibleRemainder(n: Int, rangeMax: Int) -> (divisor: Int, remainder: Int) {
  var divisorWithLargestRemainder = 1
  for d in (2..<rangeMax){
    if n % d > n % divisorWithLargestRemainder {
      divisorWithLargestRemainder = d
    }
  }
  return (divisor: divisorWithLargestRemainder, remainder: n % divisorWithLargestRemainder)
}

print(largestPossibleRemainder(20, rangeMax: 10))
